File: features/transaction/checkout/src/main/java/com/tokopedia/checkout/view/feature/emptycart/adapter/WishlistAdapter.java
package com.tokopedia.checkout.view.feature.emptycart.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.tokopedia.checkout.view.feature.emptycart.EmptyCartContract;
import com.tokopedia.checkout.view.feature.emptycart.viewholder.WishlistViewHolder;
import com.tokopedia.checkout.view.feature.emptycart.viewmodel.WishlistViewModel;
import com.tokopedia.wishlist.common.data.source.cloud.model.Wishlist;

/**
 * Created by Irfan Khoirul on 21/09/18.
 */

public class WishlistAdapter extends RecyclerView.Adapter<WishlistViewHolder> {

    private final EmptyCartContract.Presenter presenter;
    private final ActionListener actionListener;
    private final int itemWidth;

    public WishlistAdapter(EmptyCartContract.Presenter presenter, ActionListener actionListener, int itemWidth) {
        this.presenter = presenter;
        this.actionListener = actionListener;
        this.itemWidth = itemWidth;
    }

    @NonNull
    @Override
    public WishlistViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(WishlistViewHolder.LAYOUT_WISHLIST, parent, false);
        return new WishlistViewHolder(view, actionListener, itemWidth);
    }

    @Override
    public void onBindViewHolder(@NonNull WishlistViewHolder holder, int position) {
        if (presenter != null && presenter.getWishlistViewModels() != null) {
            WishlistViewModel data = presenter.getWishlistViewModels().get(position);
            holder.bindData(data);
        }
    }

    @Override
    public int getItemCount() {
        if (presenter != null && presenter.getWishlistViewModels() != null) {
            return presenter.getWishlistViewModels().size();
        }
        return 0;
    }

    public interface ActionListener {

        void onItemWishListClicked(Wishlist wishlist, int position);

    }
}

File: features/transaction/checkout/src/main/java/com/tokopedia/checkout/view/feature/emptycart2/adapter/WishlistAdapter.kt
package com.tokopedia.checkout.view.feature.emptycart2.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.tokopedia.checkout.view.feature.emptycart2.ActionListener
import com.tokopedia.checkout.view.feature.emptycart2.uimodel.WishlistItemUiModel
import com.tokopedia.checkout.view.feature.emptycart2.viewholder.WishlistItemViewHolder
import com.tokopedia.wishlist.common.data.source.cloud.model.Wishlist

/**
 * Created by Irfan Khoirul on 2019-05-20.
 */

class WishlistAdapter(val listener: ActionListener, val itemWidth: Int) : RecyclerView.Adapter<WishlistItemViewHolder>() {

    private var wishlistUiModels = ArrayList<WishlistItemUiModel>()

    fun setData(wishlistItems: List<Wishlist>) {
        for (wishlist: Wishlist in wishlistItems) {
            val wishlistUiModel = WishlistItemUiModel()
            wishlistUiModel.wishlist = wishlist
            wishlistUiModels.add(wishlistUiModel)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WishlistItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(WishlistItemViewHolder.LAYOUT, parent, false)
        return WishlistItemViewHolder(view, listener, itemWidth)
    }

    override fun getItemCount(): Int {
        return wishlistUiModels.size
    }

    override fun onBindViewHolder(holder: WishlistItemViewHolder, position: Int) {
        holder.bind(wishlistUiModels.get(position))
    }

}